core/schema/modulesource.go

```diff
@@ -2171,89 +2171,91 @@ func (s *moduleSourceSchema) moduleSourceAsModule(
 	}
 	srcInstContentHashed := src.WithDigest(digest.Digest(src.Self.Digest))
 
-	// get the runtime container, which is what is exec'd when calling functions in the module
-	mod.Runtime, err = src.Self.SDKImpl.Runtime(ctx, mod.Deps, srcInstContentHashed)
-	if err != nil {
-		return inst, fmt.Errorf("failed to get module runtime: %w", err)
-	}
-
-	// construct a special function with no object or function name, which tells
-	// the SDK to return the module's definition (in terms of objects, fields and
-	// functions)
-	getModDefCtx, getModDefSpan := core.Tracer(ctx).Start(ctx, "asModule getModDef", telemetry.Internal())
-	modName := src.Self.ModuleName
-	getModDefFn, err := core.NewModFunction(
-		getModDefCtx,
-		src.Self.Query,
-		mod,
-		nil,
-		mod.Runtime,
-		core.NewFunction("", &core.TypeDef{
-			Kind:     core.TypeDefKindObject,
-			AsObject: dagql.NonNull(core.NewObjectTypeDef("Module", "")),
-		}))
-	if err != nil {
-		getModDefSpan.End()
-		return inst, fmt.Errorf("failed to create module definition function for module %q: %w", modName, err)
-	}
-	result, err := getModDefFn.Call(getModDefCtx, &core.CallOpts{
-		Cache:          true,
-		SkipSelfSchema: true,
-		Server:         s.dag,
-		// Don't include the digest for the current call (which is a bunch of module source stuff, including
-		// APIs that are cached per-client when local sources are involved) in the cache key of this
-		// function call. That would needlessly invalidate the cache more than is needed, similar to how
-		// we want to scope the codegen cache keys by the content digested source instance above.
-		SkipCallDigestCacheKey: true,
-	})
-	if err != nil {
-		getModDefSpan.End()
-		return inst, fmt.Errorf("failed to call module %q to get functions: %w", modName, err)
-	}
-	if postCallRes, ok := dagql.UnwrapAs[dagql.PostCallable](result); ok {
-		var postCall func(context.Context) error
-		postCall, result = postCallRes.GetPostCall()
-		if postCall != nil {
-			if err := postCall(ctx); err != nil {
-				getModDefSpan.End()
-				return inst, fmt.Errorf("failed to run post-call for module %q: %w", modName, err)
+	if src.Self.SDKImpl != nil {
+		// get the runtime container, which is what is exec'd when calling functions in the module
+		mod.Runtime, err = src.Self.SDKImpl.Runtime(ctx, mod.Deps, srcInstContentHashed)
+		if err != nil {
+			return inst, fmt.Errorf("failed to get module runtime: %w", err)
+		}
+
+		// construct a special function with no object or function name, which tells
+		// the SDK to return the module's definition (in terms of objects, fields and
+		// functions)
+		getModDefCtx, getModDefSpan := core.Tracer(ctx).Start(ctx, "asModule getModDef", telemetry.Internal())
+		modName := src.Self.ModuleName
```

Comment:
@grouville you can move this line up, outside the `if` block so that on line 2258 modName stays in scope.